packages/client/src/bags/types/bagOperation.types.ts

```diff
@@ -0,0 +1,177 @@
+import type { BagItem } from './bagItem.types';
+
+export enum BagChangeReason {
+  BetterPromotionApplied = 'BetterPromotionApplied',
+  ExternalCallTimedOut = 'ExternalCallTimedOut',
+  InvalidPromocode = 'InvalidPromocode',
+  LackOfBenefits = 'LackOfBenefits',
+  NotActivePromotion = 'NotActivePromotion',
+  OnlyForFirstPurchase = 'OnlyForFirstPurchase',
+  OrderNotQualifiedForPromotion = 'OrderNotQualifiedForPromotion',
+  PromocodeAlreadyUsedByOtherCustomer = 'PromocodeAlreadyUsedByOtherCustomer',
+  PromocodeNoLongerAvailable = 'PromocodeNoLongerAvailable',
+  Unknown = 'Unknown',
+  UserNotQualifiedForPromotion = 'UserNotQualifiedForPromotion',
+}
+
+type NotificationPrice = {
+  priceInclTaxes?: number;
+};
+
+export enum ChangeType {
+  ItemDiscountChanged = 'ItemDiscountChanged',
+  ItemIsNowAvailable = 'ItemIsNowAvailable',
+  ItemIsUnavailable = 'ItemIsUnavailable',
+  ItemMerged = 'ItemMerged',
+  ItemOriginChanged = 'ItemOriginChanged',
+  ItemPriceChanged = 'ItemPriceChanged',
+  ItemQuantityChanged = 'ItemQuantityChanged',
+  NotEnoughStock = 'NotEnoughStock',
+  PromoCodeEligibilityChanged = 'PromoCodeEligibilityChanged',
+  PromotionNotApplied = 'PromotionNotApplied',
+  SelectedSaleIntentChanged = 'SelectedSaleIntentChanged',
+  ShippingCostChanged = 'ShippingCostChanged',
+  ShippingFlatRateEligibilityChanged = 'ShippingFlatRateEligibilityChanged',
+  TotalDiscountChanged = 'TotalDiscountChanged',
+}
+
+interface OperationChange {
+  changeType: ChangeType;
+  currency?: string;
+  itemId?: number;
+  newPrice?: NotificationPrice;
+  newValue?: string;
+  oldCurrency?: string;
+  oldPrice?: NotificationPrice;
+  oldValue?: string;
+  productId?: number;
```

Comment:
Can this be `Product['result']['id'] instead?`

Reply:
Done!